src/commands/lambda/flare.ts

```diff
@@ -40,105 +49,170 @@ export class LambdaFlareCommand extends Command {
     this.context.stdout.write(flareRenderer.renderLambdaFlareHeader(this.isDryRun))
 
     // Validate function name
+    const errorMessages = []
     if (this.functionName === undefined) {
-      this.context.stderr.write(commonRenderer.renderError('No function name specified. [-f,--function]'))
-
-      return 1
+      errorMessages.push(commonRenderer.renderError('No function name specified. [-f,--function]'))
     }
 
     // Validate region
-    let errorFound = false
-    const region = getRegion(this.functionName) ?? this.region ?? process.env[AWS_DEFAULT_REGION_ENV_VAR]
+    const region = getRegion(this.functionName ?? '') ?? this.region ?? process.env[AWS_DEFAULT_REGION_ENV_VAR]
     if (region === undefined) {
-      this.context.stderr.write(commonRenderer.renderNoDefaultRegionSpecifiedError())
-      errorFound = true
+      errorMessages.push(commonRenderer.renderNoDefaultRegionSpecifiedError())
     }
 
     // Validate Datadog API key
     this.apiKey = process.env[CI_API_KEY_ENV_VAR] ?? process.env[API_KEY_ENV_VAR]
     if (this.apiKey === undefined) {
-      this.context.stderr.write(
+      errorMessages.push(
         commonRenderer.renderError(
           'No Datadog API key specified. Set an API key with the DATADOG_API_KEY environment variable.'
         )
       )
-      errorFound = true
     }
 
     // Validate case ID
     if (this.caseId === undefined) {
-      this.context.stderr.write(commonRenderer.renderError('No case ID specified. [-c,--case-id]'))
-      errorFound = true
+      errorMessages.push(commonRenderer.renderError('No case ID specified. [-c,--case-id]'))
     }
 
     // Validate email
     if (this.email === undefined) {
-      this.context.stderr.write(commonRenderer.renderError('No email specified. [-e,--email]'))
-      errorFound = true
+      errorMessages.push(commonRenderer.renderError('No email specified. [-e,--email]'))
     }
 
-    if (errorFound) {
+    // Exit if there are errors
+    if (errorMessages.length > 0) {
+      this.context.stderr.write(errorMessages.join('') + '\n')
+
       return 1
     }
+    assert(this.functionName !== undefined)
```

Comment:
What's the reason of changing the validation on this PR? This doesn't seem related to the PR description changes.

Reply:
Also, why are we asserting? Isn't the first validaton check supposed to do this?

Reply:
This change is not necessary for this PR, but I think this solution would be more readable and take up less lines of code. Logically, it does the same thing as it did before.
This file, for example, uses the same approach: https://github.com/DataDog/datadog-ci/blob/3d41ad6db1fed0f0bcc6216b49d534961673fd8a/src/commands/synthetics/utils.ts#L567-L576

I am asserting because I am trying to force the type of `this.functionName` to be `string` instead of `string | undefined`. Do you think there is a better approach, such as doing `this.functionName ?? ''` instead?

Reply:
There are better ways of forcing a type in TypeScript.
Either you type cast, type convert, or simply use the non-null assertion operator. Why not do that instead of assert?

Nice, sounds good, let's do it in another PR that focuses only in refactoring this!

Normally, reviewers expect to find changes in the logic in which the description is referring to. Those changes add value to the whole project, but are not directly tied to the PR description. So yeah, generally, if a change is not related, we would expect a separate PR.

Reply:
Understood. Logged a ticket for improving flag validation!